app/src/test/java/com/invisibleteam/goinvisible/mvvm/edition/EditActivityTest.java

```diff
@@ -0,0 +1,54 @@
+package com.invisibleteam.goinvisible.mvvm.edition;
+
+import android.content.Intent;
+
+import com.invisibleteam.goinvisible.BuildConfig;
+import com.invisibleteam.goinvisible.model.ImageDetails;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.Robolectric;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.RuntimeEnvironment;
+import org.robolectric.annotation.Config;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+@RunWith(RobolectricTestRunner.class)
+@Config(constants = BuildConfig.class, sdk = 21)
+@SuppressFBWarnings("NP_NONNULL_PARAM_VIOLATION")
+public class EditActivityTest {
+
+    private EditActivity activity;
+
+    @Test
+    public void whenActivityIsStartedWithIntentAndIntentContainsData_thenDataAreFetchedCorrectly() {
+        //given
+        ImageDetails imageDetails = new ImageDetails("Path", "Name");
+        Intent editActivityIntent = EditActivity.buildIntent(RuntimeEnvironment.application, imageDetails);
```

Comment:
RuntimeEnvironment.application could be extracted as a class member (context) and initiated on @Before